DAQ/src/ArtBinaryPacketsFromDigis_module.cc

```diff
@@ -0,0 +1,1113 @@
+    printf("[ArtBinaryPacketsFromDigis::printTrackerData] ADC12  	: %i \n", (int)trkData.ADC12);
+    printf("[ArtBinaryPacketsFromDigis::printTrackerData] ADC13 	: %i \n", (int)trkData.ADC13());
+    printf("[ArtBinaryPacketsFromDigis::printTrackerData] ADC14 	: %i \n", (int)trkData.ADC14());
+    printf("[ArtBinaryPacketsFromDigis::printTrackerData] unused1 	: %i \n", (int)trkData.unused1);
+    printf("[ArtBinaryPacketsFromDigis::printTrackerData] PreprocessingFlags : %i \n", (int)trkData.PreprocessingFlags);
+
+
+  }
+
+  void   ArtBinaryPacketsFromDigis::printCalorimeterData(CaloDataPacket const& caloData) {
+    CalorimeterDataPacket packet = caloData.dataPacket;
+    CalorimeterBoardID    boardId = caloData.boardID;
+    size_t      nHits = caloData.hitPacketVec.size();
+    printf("[ArtBinaryPacketsFromDigis::printCaloData] START calorimeter-data print \n");
+    printf("[ArtBinaryPacketsFromDigis::printCaloData] NumberofHits        : %i \n", (int)packet.NumberOfHits);
+    printf("[ArtBinaryPacketsFromDigis::printCaloData] BoardID             : %i \n", (int)boardId.BoardID);
+    printf("[ArtBinaryPacketsFromDigis::printCaloData] ChannelStatusFlagsA : %i \n", (int)boardId.ChannelStatusFlagsA);
+    printf("[ArtBinaryPacketsFromDigis::printCaloData] ChannelStatusFlagsB : %i \n", (int)boardId.ChannelStatusFlagsB);
+    printf("[ArtBinaryPacketsFromDigis::printCaloData] unused              : %i \n", (int)boardId.unused);
+    printf("[ArtBinaryPacketsFromDigis::printCaloData] NHits               : %i \n", (int)nHits);
+
+    for (size_t i = 0; i < nHits; ++i) {
+      CalorimeterHitReadoutPacket const& hit = caloData.hitPacketVec[i];
+      printf("[ArtBinaryPacketsFromDigis::printCaloData] hit : %i \n", (int)i);
+      printf("[ArtBinaryPacketsFromDigis::printCaloData] ChannelNumber : %i \n", (int)hit.ChannelNumber);
+      printf("[ArtBinaryPacketsFromDigis::printCaloData] DIRACA        : %i \n", (int)hit.DIRACA);
+      printf("[ArtBinaryPacketsFromDigis::printCaloData] DIRACB        : %i \n", (int)hit.DIRACB);
+      printf("[ArtBinaryPacketsFromDigis::printCaloData] ErrorFlags    : %i \n", (int)hit.ErrorFlags);
+      printf("[ArtBinaryPacketsFromDigis::printCaloData] Time          : %i \n", (int)hit.Time);
+      printf("[ArtBinaryPacketsFromDigis::printCaloData] NumberOfSamples : %i \n", (int)hit.NumberOfSamples);
+      printf("[ArtBinaryPacketsFromDigis::printCaloData] IndexOfMaxDigitizerSample : %i \n", (int)hit.IndexOfMaxDigitizerSample);
+    }
+
+  }
+
+  void   ArtBinaryPacketsFromDigis::fillTrackerDataStream(raw_data_list_t& dataStream,
+							  tracker_data_block_t const& trackerData) {
+
+    auto sz = sizeof(DataBlockHeader);
+    //check that the trkDataBlock is not empty
+
+    assert(trackerData.first.PacketCount == 0 || trackerData.first.PacketCount == 2); // Tracker DataBlocks have 0 or 2 DataPackets
+    if (trackerData.first.PacketCount > 0) {
```

Comment:
Remember that asserts are ignored in our prof builds.  Is that the intended behaviour?